bigquery_etl/stripe/__init__.py

```diff
@@ -271,25 +174,19 @@ def stripe_import(
         handle = sys.stdout.buffer
     with handle as file_obj:
         has_rows = False
-        if resource:
-            filtered_schema = FilteredSchema(resource)
-            for row in _get_rows(api_key, after_date, before_date, resource):
-                if format_resources or (table and api_key):
-                    row = filtered_schema.format_row(row, strict=strict_schema)
-                file_obj.write(ujson.dumps(row).encode("UTF-8"))
-                file_obj.write(b"\n")
-                has_rows = True
-        elif report_type:
-            root = get_rooted_schema(report_type)
-            columns = [f.name for f in root.fields]
-            for row in _get_report_rows(
-                api_key, after_date, before_date, report_type, columns
-            ):
-                file_obj.write(row)
-                file_obj.write(b"\n")
-                # stripe api enforces that data is available for the requested range, so
-                # allow headers row to count for has_rows, because it means the report
-                # is complete, and valid reports may be empty, e.g. when pulled daily
+        path = Path(__file__).parent / f"{report_type}.schema.json"
+        root = bigquery.SchemaField.from_api_repr(
+            {"name": "root", "type": "RECORD", "fields": ujson.loads(path.read_text())}
+        )
+        columns = [f.name for f in root.fields]
+        has_headers = False
+        for row in _get_report_rows(
+            api_key, after_date, before_date, report_type, columns
+        ):
+            file_obj.write(row)
+            if not has_headers:
+                has_headers = True
+            else:
                 has_rows = True
```

Comment:
I'm not sure, I think it had to do with `--allow-empty` being added. I've instead updated this to remove `has_rows` and `--allow-empty` entirely, because they don't make sense when reports throw an API exception if requested before data is complete.